### PR TITLE
[Enhancement](Es): Support config like whether push down to es

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Support config like whether push down to es and refactor some code

Like transform to wildcard query and push down to es, this increases the cpu consumption of the es, I add a switch control it.

## Checklist(Required)

* [x] Does it affect the original behavior
* [x] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java [hunk 1]
```diff
@@ -213,167 +194,6 @@ public static JSONObject getMappingProps(String sourceIndex, String indexMapping
         return properties;
     }
 
-    private static QueryBuilder toCompoundEsDsl(Expr expr, List<Expr> notPushDownList,
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java:216]: move to QueryBuilders

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java [hunk 2]
```diff
@@ -38,6 +59,238 @@
...
+        // Replace col with col.keyword if mapping exist.
+        column = fieldsContext.getOrDefault(column, column);
+        if (expr instanceof BinaryPredicate) {
+            return parseBinaryPredicate(expr, opCode, column);
+        }
+        if (expr instanceof IsNullPredicate) {
+            return parseIsNullPredicate(expr, column);
+        }
+        if (expr instanceof LikePredicate) {
+            if (!likePushDown) {
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java:252]: Judge whether push down like

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java [hunk 3]
```diff
@@ -38,6 +59,238 @@
  */
 public final class QueryBuilders {
 
+    /**
+     * Generate dsl from compound expr.
+     **/
+    private static QueryBuilder toCompoundEsDsl(Expr expr, List<Expr> notPushDownList,
+            Map<String, String> fieldsContext, boolean likePushdown) {
```
Comment (SaintBacchus) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java:66]: should we use a `BuilderOptions` or `BuilderContext` to wrap the `boolean likePushdown` if it has more option later?

**Review by qidaye — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java (hunk 3 above)
Comment (qidaye) [reply fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java:66]: > should we use a `BuilderOptions` or `BuilderContext` to wrap the `boolean likePushdown` if it has more option later?

Good idea.

Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java:66]: Okay, I will modify it.

**Comment by SaintBacchus** [comment]
LGTM

**Review by qidaye — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**